### PR TITLE
Merge Hdrp/staging

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR

- Hdrp/depth offset shader graph (#3114) 
- Recorder implementation for HDRP (#3111) 
- Hdrp/fix densityvolume handles (#3124) 
- Fixed calls to InitBuiltinData in VFX graph shaders (#3147)
- Hdrp/fix lightgizmo multi edition (#3142) 
- Move scene color to R11G11B10 (#3143) 
- Fix unwanted commented migration version (#3152)
- Don't allocate scattering textures if not supported (#3166)
- Delete optimal group size define (#3175) 
- Zero out stencil bit after they are not needed anymore (#3195) 
- Disable SceneSelectionPass in shader graph preview (#3205)
- Disable light normalization in preferences
- Control punctual light and area light shadow atlas separately (#3115)
- Disable edge leak fixup by default (#3211)
- Disable stencil and fix smaa for xr (first pass) (#3207)
- Added SMAA support

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: What did you setup?

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?

**Halo Effect**: None, Low, Medium, High?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs [hunk 1]
```diff
@@ -208,7 +208,14 @@ static void DrawGeneralAdvancedContent(SerializedHDLight serialized, Editor owne
             using (new EditorGUI.DisabledScope(!HDUtils.hdrpSettings.supportLightLayers))
             {
                 var renderingLayerMask = serialized.serializedLightData.renderingLayerMask.intValue;
-                var lightLayer = HDAdditionalLightData.RenderingLayerMaskToLightLayer(renderingLayerMask);
+                int lightLayer;
+                if (serialized.serializedLightData.renderingLayerMask.hasMultipleDifferentValues)
+                {
+                    EditorGUI.showMixedValue = true;
+                    lightLayer = 0;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs]: check with Remi

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs [hunk 2]
```diff
@@ -449,21 +467,30 @@ static void ConvertLightIntensity(LightUnit oldLightUnit, LightUnit newLightUnit
...
+                    else
+                        EditorGUILayout.PropertyField(serialized.settings.color, s_Styles.color);
                 }
                 else
                     EditorGUILayout.PropertyField(serialized.settings.color, s_Styles.color);
+                
+                if (changes.changed && HDRenderPipelinePreferences.lightColorNormalization)
+                {
+                    Vector4 ldrColor = Vector4.Max(serialized.settings.color.colorValue, Vector4.one * 0.0001f);
+                    serialized.settings.color.colorValue = (Vector4)(ldrColor / ColorUtils.Luminance(ldrColor));
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs]: It required to have a test for black, max(0.0001f can introduce error right ?)

File: com.unity.render-pipelines.high-definition/Editor/Material/Lit/LitUI.cs [hunk 3]
```diff
@@ -887,6 +887,15 @@ protected void DoEmissiveGUI(Material material)
                         EditorGUI.BeginChangeCheck();
                         {
                             DoEmissiveTextureProperty(material, emissiveColorLDR);
+                            if (HDRenderPipelinePreferences.materialEmissionColorNormalization)
+                            {
+                                // When enabling the material emission color normalization the ldr color might not be normalized,
+                                // so we need to update the emissive color
+                                if (!Mathf.Approximately(ColorUtils.Luminance(emissiveColorLDR.colorValue), 1))
+                                    updateEmissiveColor = true;
+                                Vector4 ldrColor = Vector4.Max(emissiveColorLDR.colorValue, Vector4.one * 0.0001f);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Lit/LitUI.cs]: code must be share! :)

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/Lit/LitUI.cs]: is it apply to unlit and layered lit ?

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 4]
```diff
@@ -400,8 +408,8 @@ void InitializeRenderTextures()
...
             m_SharedRTManager.InitSharedBuffers(m_GbufferManager, m_Asset.currentPlatformRenderPipelineSettings, m_Asset.renderPipelineResources);
 
-            m_CameraColorBuffer = RTHandles.Alloc(Vector2.one, filterMode: FilterMode.Point, colorFormat: GraphicsFormat.R16G16B16A16_SFloat, enableRandomWrite: true, useMipMap: false, xrInstancing: true, useDynamicScale: true, name: "CameraColor");
-            m_CameraSssDiffuseLightingBuffer = RTHandles.Alloc(Vector2.one, filterMode: FilterMode.Point, colorFormat: GraphicsFormat.B10G11R11_UFloatPack32, enableRandomWrite: true, xrInstancing: true, useDynamicScale: true, name: "CameraSSSDiffuseLighting");
+            m_CameraColorBuffer = RTHandles.Alloc(Vector2.one, filterMode: FilterMode.Point, colorFormat: GetColorBufferFormat(), enableRandomWrite: true, useMipMap: false, xrInstancing: true, useDynamicScale: true, name: "CameraColor");
+            m_CameraSssDiffuseLightingBuffer = RTHandles.Alloc(Vector2.one, filterMode: FilterMode.Point, colorFormat: GetColorBufferFormat(), enableRandomWrite: true, xrInstancing: true, useDynamicScale: true, name: "CameraSSSDiffuseLighting");
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: here we have lose an optimization. Diffuse lighting must stay 101111 all the time.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipelineResources/HDRenderPipelineResources.asset [hunk 5]
```diff
@@ -34,6 +34,7 @@ MonoBehaviour:
     applyDistortionCS: {fileID: 7200000, guid: 2fa6c0e3fe6dc3145a4156f21913fe5c, type: 3}
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipelineResources/HDRenderPipelineResources.asset]: This must be removed! applyDistortionCS

File: com.unity.render-pipelines.high-definition/Runtime/Utilities/HDRenderPipelinePreferences.cs [hunk 6]
```diff
@@ -38,10 +38,36 @@ public static bool sceneViewStopNaNs
             }
         }
 
+        static bool s_LightColorNormalization = false;
+        public static bool lightColorNormalization
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Utilities/HDRenderPipelinePreferences.cs]: what is this in preference ?

File: com.unity.render-pipelines.core/Runtime/Utilities/CameraCaptureBridge.cs [hunk 7]
```diff
@@ -5,20 +5,17 @@
...
 using System.Reflection;
 #else
 using UnityEditor.Recorder;
 #endif
 #endif
-using UnityEngine.Rendering;
 
 
-namespace UnityEngine.Rendering.LWRP
+namespace UnityEngine.Rendering
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/Runtime/Utilities/CameraCaptureBridge.cs]: is it expected in this PR ?

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipelineResources/HDRenderPipelineResources.asset (hunk 5 above)
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipelineResources/HDRenderPipelineResources.asset]: I'll check asap.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs (hunk 4 above)
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Will fix this as well

**Comment by sebastienlagarde** [comment]
Katana is green on the PR Disable stencil and fix smaa for xr (first pass) (#3207) which is the latest PR with latest code. No need to relaunch katana here.

**Pull request merged**